src/strspan.rs

```diff
@@ -23,6 +23,30 @@ impl<'a> From<&'a str> for StrSpan<'a> {
     }
 }
 
+impl PartialEq<str> for StrSpan<'_> {
```

Comment:
Do you really need this many comparisons? Can you provide a use case for them?

Reply:
Well you need to implement `PartialEq` commutatively and for cases where they are unmatching numbers of reference.
We could remove two of the implementations and require the user to dereference themselves but I believe that traits should be [eagerly implemented](https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits) in general.

Reply:
I think we should remove the second trait (`&&str` one) and the third one.
Unless you can provide an example where they are actually needed.

Reply:
I'm using the second in multiple places like so:
```rust
match tkn {
    Token::Attribute {
        prefix,
        local,
        value,
        ..
    } if prefix.is_empty() && local == "name" => {
       // do something
    }
 ```
 
 Its commutative version being the fourth.
 
 First and third would be needed in a similar case if I'm only peeking the Tokenizer and hence only have a `&Token` that I don't want to move out of for whatever reason.
 
So if two must go I'd advocate for 1st and 3rd; which would have the advantage of not breaking my current code.

Reply:
Ok, let's keep all of them. Hopefully it will be worth it.

Reply:
Ok :)